src/main/webapp/WEB-INF/pages/productList.jsp

```diff
@@ -40,10 +40,37 @@
                 </td>
                 <td class="price">
                     <a href="${pageContext.servletContext.contextPath}/products/prices/${product.id}" target="_blank">
-                        <fmt:formatNumber value="${product.price}" type="currency" currencySymbol="${product.currency.symbol}"/>
+                        <fmt:formatNumber value="${product.price}" type="currency"
+                                          currencySymbol="${product.currency.symbol}"/>
                     </a>
                 </td>
             </tr>
         </c:forEach>
     </table>
+
+    <p>
+        Recently viewed
+    </p>
+    <ul style="text-align: left; padding-inline-start: 0px;">
+        <c:forEach var="product" items="${recentlyViewedProducts}">
+            <li style="display: inline-block;">
+                <table>
+                    <tr>
+                        <td style="text-align: center;">
+                            <img class="product-tile" src="${product.imageUrl}">
+                            <div>
+                                <a href="${pageContext.servletContext.contextPath}/products/${product.id}">
+                                        ${product.description}
+                                </a>
+                            </div>
+                            <div>
+                                <fmt:formatNumber value="${product.price}" type="currency"
+                                                  currencySymbol="${product.currency.symbol}"/>
+                            </div>
+                        </td>
+                    </tr>
+                </table>
+            </li>
+        </c:forEach>
+    </ul>
```

Comment:
Move to separate tag so it can be reused

Reply:
Ok